R/parser.R

```diff
@@ -4,6 +4,281 @@
+      issue_status = append(issue_comment[["data.bug_status"]][[i]],NA)[1],
+      issue_resolution = append(issue_comment[["data.resolution"]][[i]],NA)[1],
+      issue_components = append(issue_comment[["data.component"]][[i]],NA)[1],
+      issue_description = append(issue_comment[["data.long_desc"]][[i]][["thetext"]][[1]],NA)[1],
+      issue_classification = append(issue_comment[["data.classification"]][[i]],NA)[1], ##### ADDING
+
+      issue_created_datetimetz = append(issue_comment[["data.creation_ts"]][[i]],NA)[1],
+      issue_updated_datetimetz = append(issue_comment[["data.delta_ts"]][[i]],NA)[1],
+
+      issue_assignee_id = append(issue_comment[["data.assigned_to"]][[i]][["__text__"]],NA)[1],
+      issue_assignee_name = append(issue_comment[["data.assigned_to"]][[i]][["name"]],NA)[1],
+
+      issue_reporter_id = append(issue_comment[["data.reporter"]][[i]][["__text__"]],NA)[1],
+      issue_reporter_name = append(issue_comment[["data.reporter"]][[i]][["name"]],NA)[1],
+
+      issue_target_milestone = append(issue_comment[["data.target_milestone"]][[i]],NA)[1],
+      issue_rep_platform = append(issue_comment[["data.rep_platform"]][[i]],NA)[1],
+      issue_status_whiteboard = append(issue_comment[["data.status_whiteboard"]][[i]],NA)[1],
+
+      issue_version = append(issue_comment[["data.version"]][[i]],NA)[1],
+      issue_severity = append(issue_comment[["data.bug_severity"]][[i]],NA)[1],
+      issue_priority = append(issue_comment[["data.priority"]][[i]],NA)[1],
+      issue_op_system = append(issue_comment[["data.op_sys"]][[i]],NA)[1],
+      issue_product = append(issue_comment[["data.product"]][[i]],NA)[1]
+    )
+
+    # Comments
+    # For each issue, it may have 0 or more comments. Parse them
+    # in a separate table if the comments argument is true.
+    if (comments == TRUE) {
+      root_of_comments_list <- json_issue_comments[["data.long_desc"]][[i]]
+
+      #If root_of_comments_list does not exist, then this is an issue only json, skip parsing
+      if(length(root_of_comments_list) > 0){
+        comments_list <- json_issue_comments[["data.long_desc"]][[i]]
+
+        # Even on a json with comments, some issues may not have comments, check if comments exist:
+        if(length(comments_list[["commentid"]]) > 1){
+          # Parse all comments into issue_comments
+          issue_comments <- rbindlist(bugzilla_parse_comment(comments_list))
+          # Add issue_key column to the start of the table
+          issue_comments <- cbind(data.table(issue_key=issue_key),issue_comments)
+          all_issues_comments[[i]] <- issue_comments
+        }
+      }
+    }
+  }
+
+  all_issues <- rbindlist(all_issues,fill=TRUE)
+  all_issues_comments <- rbindlist(all_issues_comments,fill=TRUE)
+
+  parsed_issues_comments <- list()
+  parsed_issues_comments[["issues"]] <- all_issues
+  parsed_issues_comments[["comments"]] <- all_issues_comments
+
+  return(parsed_issues_comments)
```

Comment:
I found the return of the function to be confusing. If comments flag is off, a list of 2 elements is returned, where the second is empty, but if the flag is on, then both tables are present. 

My suggestion is to simply return the tables themselves. If comments is off, return the data table of issues, if comments is on, then return a single table of issues + comments. I believe this is how Jira Issue behaves, no? It's been awhile.

Reply:
Hi Carlos, I think that parse_jira() returns the tables similarly to how I did it. The function docs for parse_jira() says the following:
https://github.com/sailuh/kaiaulu/blob/716a46e546d5293f3b2856cace62311c95aae977/R/parser.R#L289-L290

And has a similar return:
https://github.com/sailuh/kaiaulu/blob/716a46e546d5293f3b2856cace62311c95aae977/R/parser.R#L385-L389

Reply:
I see. So that's where you got this from. Unfortunately, `parse_jira` is not the best code to reference as a model beyond the nomenclature. This is because the downloader (which you are implementing) is not implemented in Kaiaulu: 

http://itm0.shidler.hawaii.edu/kaiaulu/articles/download_jira_data.html 

So there is less control over how the data is stored. That being said, the change I am requesting here can likely be applied to the function too. Could you make an issue for that and assign it to me?

Reply:
Alright, I will do that.

Reply:
Were the returns fixed on the more recent PRs? Having a bit of a hard time assessing at this point since the PR grew big!